General Fixes

dmportella opened the pull request:
Implemented updates to instrument tracking to address exceptions thrown by missing config containers and added handling for BmpTransmogrifyException. Resolved the issues reported in GitHub issue #178.

dmportella committed:
ADDED safer getter for track Instrument

dmportella committed:
REMOVED defunct tolist

dmportella committed:
ADDED INotifyPropertyChanged

This will make the ui component work properly without having to set the data context / binding all the time

dmportella committed:
CHANGES added correct handling of data binding

Change UpdateList to use the items refresh command isntead of reseting the bind with a new reference to the list as defunct and not required

removed bmpmaestro events that are no longer required

this change fixes https://github.com/BardMusicPlayer/BardMusicPlayer/issues/178

Meowchestra commented:
@dmportella Can you remove the ``e`` variable in the exception catch? No build warnings. 🙏😸

![link](https://i.imgur.com/2LhQNbh.png)

dmportella commented:
> @dmportella Can you remove the `e` variable in the exception catch? No build warnings. 🙏😸
> 
> ![link](https://camo.githubusercontent.com/f63614b3ae3d5ed6b0499cc48880852c15a1f3779165dee680733289a8f9a20d/68747470733a2f2f692e696d6775722e636f6d2f324c68514e62682e706e67)

yep will do there is a few more changes coming in

dmportella committed:
Fixed bug in track down as it was allowing tracks to be below 1

dmportella committed:
ADDED bard list view model to allow data binding to be handled once

view model added
set on XAML now instead of code
still using on performer changed to set the initial and future list changes but now events are fired properly

dmportella committed:
removed defunct variable

dmportella commented:
This concludes this bout of changes it closes  #178 

Also it raises another technical debt item - handling data binding

A lot of the controls and classes should really use the INotify family of interfaces for the data binding to be handled properly, as it stand numerous properties and events are fired countless times when ItemSources and DataContext are been reset every time.

Specially with so many thread / parallel calls in code this will eventually slow down and more the UI not as responsive as the work progresses.

Anyway might not a big issue with people PC getting fast anyway  . . . 

Some other notes in the BardListView a lot of the events from BmpMaestro and BmpSeer could be handled using INotify instead (does mean the existing event are not required its just for all intents of purposes of data binding) I have kept the current ones in the file that are still required since i have not change the underlying classes to use data binding practices.
